packaging/deb/build.sh

```diff
@@ -0,0 +1,116 @@
+### Check for basic requirements
+################################################################################
+
+if ! which git >/dev/null 2>&1; then
+    echo "git not installed"
+    exit 1
+fi
+if ! git rev-parse >/dev/null 2>&1; then
+    echo "not a git repository"
+    exit 1
+fi
+if [ "$(git rev-parse --show-toplevel)" != "$SRC_DIR" ]; then
+    echo "$SRC_DIR is not a git repository"
+    exit 1
+fi
+if ! which python >/dev/null 2>&1; then
+    echo "python not installed"
+    exit 1
+fi
+if ! git diff-index --quiet HEAD >/dev/null 2>&1; then
+    echo "git index is dirty - either stash or commit your changes"
+    exit 1
+fi
+if ! which docker >/dev/null 2>&1; then
+    echo "docker not installed"
+    exit 1
+fi
+
+################################################################################
+# Read envvars
+################################################################################
+
+if [ -z "$DOCKER_BASE" ]; then
+    echo "DOCKER_BASE is a required environment variable"
+    exit 1
+fi
+echo "DOCKER_BASE: $DOCKER_BASE"
+DOCKER_BUILD_ID="caffe-nv-debuild"
+DOCKER_BASE_TAG="caffe-nv-debuild-base"
+docker tag ${DOCKER_BASE} ${DOCKER_BASE_TAG}
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libcudart.so ' | wc -l")" -ne 1 ]]; then
+    echo "CUDA is not installed on this image"
+    exit 1
+fi
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libcudnn.so ' | wc -l")" -ne 1 ]]; then
+    echo "cuDNN is not installed on this image"
+    exit 1
+fi
+if [[ "$(docker run --rm ${DOCKER_BASE_TAG} bash -c "ldconfig -p | grep 'libnccl.so ' | wc -l")" -ne 1 ]]; then
+    echo "NCCL is not installed on this image"
+    exit 1
+fi
+
+if [ -z "$DEBIAN_REVISION" ]; then
+    echo ">>> Using default DEBIAN_REVISION (set the envvar to override)"
+    OS_NAME=$(docker run --rm ${DOCKER_BASE_TAG} sh -c ". /etc/os-release && echo \$ID\$VERSION_ID")
+    CUDA_VERSION=$(docker run --rm ${DOCKER_BASE_TAG} bash -c "readlink /usr/local/cuda-*/lib*/libcudart.so | sort -u | cut -c14-")
+    DEBIAN_REVISION=1${OS_NAME}+cuda${CUDA_VERSION}
+fi
+echo "DEBIAN_REVISION: $DEBIAN_REVISION"
+
+################################################################################
+# Calculate versions
+################################################################################
+
+DESCRIBE_VERSION=$(git describe)
+GIT_TAG=$(git describe --abbrev=0)
+UPSTREAM_VERSION=${DESCRIBE_VERSION:1}
+if [[ "$GIT_TAG" == *"-"* ]]; then
+    # Replace the first dash with a tilde
+    UPSTREAM_VERSION=$(echo $UPSTREAM_VERSION | sed '0,/-/{s/-/~/}')
+fi
+# Replace the first dash with a plus
+UPSTREAM_VERSION=$(echo $UPSTREAM_VERSION | sed '0,/-/{s/-/\+/}')
+# Replace all dashes with dots
+UPSTREAM_VERSION=$(echo $UPSTREAM_VERSION | sed 's/-/\./g')
+echo UPSTREAM_VERSION: $UPSTREAM_VERSION
+DEBIAN_VERSION=${UPSTREAM_VERSION}-${DEBIAN_REVISION}
+echo DEBIAN_VERSION: $DEBIAN_VERSION
+
```

Comment:
Oh yeah, that's a bit cleaner.
http://www.tldp.org/LDP/abs/html/parameter-substitution.html#EXPREPL1